dynamiqs/sparse.py

```diff
@@ -0,0 +1,202 @@
+import functools
+
+import jax
+import jax.numpy as jnp
+from jaxtyping import Array
+
+
+class SparseDIA:
+    def __init__(self, diags: Array, offsets: tuple[int]):
+        self.offsets = offsets
+        self.diags = diags
+
+
+    def to_dense(self) -> Array:
```

Comment:
I think we want to duplicate this function to have a `dq.to_dense` in the public API (similarly to `dq.to_sparse`). So it's essentially the same as this one, but not as a class method. Both should have the same content, so you can simply redirect `SparseDIA.to_dense` to `dq.to_dense`.